docs/design/OCI Store Cache Worker.md

```diff
@@ -0,0 +1,191 @@
+            "Issuer": "CN=ratify-bats-test,O=Notary,L=Seattle,ST=WA,C=US",
+            "SN": "CN=ratify-bats-test,O=Notary,L=Seattle,ST=WA,C=US"
+          },
+          "artifactType": "application/vnd.cncf.notary.signature"
+        }
+      ],
+      "artifactType": "org.example.sbom.v0"
+    }
+  ]
+}
+```
+
+## User Scenarios
+
+### Using multi-verifier
+
+1. Multiple different verifiers are executed for the same subject
+2. Multiple same verifiers with different version executed for the same subject
+
+### Multi-tenancy
+
+1. reusing OCI artifacts
+
+## Purpose
+
+- Resolve OCI Store race condition in multi-verifier scenarios
+- Ensure deduplication of OCI artifacts hold in local environment
+
+## Data Flow
+
+```mermaid
+%% Initialize diagram configuration
+%%{init: {'theme': 'base', 'themeVariables': { 'primaryColor': '#ffcc00', 'edgeLabelBackground':'#f8f8f8', 'tertiaryColor': '#fff'}}}%%
+flowchart TD
+    %% Define styles
+    classDef subProcess fill:#ffcc00,stroke:#333,stroke-width:2px;
+
+    %% Define components
+    subgraph VerifierSubsystem [Verifier Subsystem]
+        GetSubjectDescriptor[Get Subject Descriptor]
+        GetSubjectReferenceManifest[Get Subject Reference Manifest]
+        GetReferrerArtifact[Get Blob Content]
+        PerformVerification[Perform Verification]
+        ReturnVerifierResult[Return Verifier Result]
+        MemoryBasedStore[Memory Based Store]
+        OCIStoreCache[OCI Store Cache]
+        OCIStoreCacheWorker[Cache Worker]
+        CacheProvider[Cache Provider]
+        ORASFetch1[ORAS Fetch]
+        ORASFetch2[ORAS Fetch]
+        
+    end
+
+    %% Define relationships
+    GetSubjectDescriptor -->|Store operation| GetSubjectReferenceManifest
+    GetSubjectReferenceManifest -->|Store operation| Cache_Hit_1
+    Cache_Hit_1{Cache Hit?}
+    Cache_Hit_1 -->|Hit| GetReferrerArtifact
+    Cache_Hit_1 -->|Miss| ORASFetch1
+    ORASFetch1 -->|Fetch| MemoryBasedStore
+    GetReferrerArtifact --> Cache_Hit_2
+    Cache_Hit_2{Cache Hit?}
+    Cache_Hit_2 -->|Hit| PerformVerification
+    Cache_Hit_2 -->|Miss| ORASFetch2
+    ORASFetch2 -->|Fetch| MemoryBasedStore
+    PerformVerification --> ReturnVerifierResult
+    OCIStoreCache --> CacheProvider
+    CacheProvider --> OCIStoreCache
+
+    %% Highlight subprocesses
+    GetReferrerArtifact:::subProcess
+    ORASFetch1:::subProcess
+    ORASFetch2:::subProcess
+    PerformVerification:::subProcess
+
+    %% Add connections to the cache
+    Cache_Hit_1 -.-> OCIStoreCacheWorker
+    MemoryBasedStore -.-> GetReferrerArtifact
+    Cache_Hit_2 -.-> OCIStoreCacheWorker
+    MemoryBasedStore -.-> PerformVerification
+    MemoryBasedStore -.-> OCIStoreCacheWorker
+    OCIStoreCacheWorker -.-> CacheProvider
+    
+
+    %% Define external components
+    classDef external fill:#f0f0f0,stroke:#333,stroke-width:2px;
+    OCIStoreCacheWorker:::external
+    MemoryBasedStore:::external
+    OCIStoreCache:::external
+    CacheProvider:::external
+```
+
```

Comment:
wonder if we could have a sequence diagram for a request, especially for the data flow among Memory based store/cache workder/cache provider/OCI store Cache.

Reply:
Would improve based on the current diagram